modules/publication/ajax/FileUpload.php

```diff
@@ -0,0 +1,879 @@
+    $uid   = $user->getId();
+    $today = date('Y-m-d');
+    // insert the titleRaw to avoid double escaping
+    $fields = array(
+               'UserID'             => $uid,
+               'Title'              => $titleRaw,
+               'Description'        => $desc,
+               'LeadInvestigatorID' => $leadInvID,
+               'DateProposed'       => $today,
+              );
+
+    $db->insert('publication', $fields);
+    $pubID = $db->getLastInsertId();
+
+    try {
+        // process files
+        processFiles($pubID);
+        // INSERT INTO publication_collaborator
+        insertCollaborators($pubID);
+        // INSERT INTO publication_users_edit_perm_rel
+        insertEditors($pubID);
+        // INSERT INTO publication_keyword
+        insertKeywords($pubID);
+        // INSERT INTO publication_parameter_type_rel
+        insertVOIs($pubID);
+        // INSERT INTO publication_users_edit_perm_rel
+    } catch (Exception $e) {
+        cleanup($pubID);
+        showError($e->getMessage());
+    }
+
+    notify($pubID, 'submission');
+}
+
+/**
+ * Function for storing files and inserting file meta data into database
+ *
+ * @param int $pubID ID of Publication in publication table
+ *
+ * @return null
+ */
+function processFiles($pubID)
+{
+    if (empty($_FILES)) {
+        return;
+    }
+    $db     = \Database::singleton();
+    $config = \NDB_Config::singleton();
+
+    $publicationPath = $config->getSetting('publication_uploads');
+
+    if (!isset($publicationPath)) {
+        showError(
+            "Error! Publication path is not set in Loris Settings!"
+        );
+    }
+
+    if (!file_exists($publicationPath)) {
+        showError(
+            "Error! The upload folder '$publicationPath' does not exist!'"
+        );
+    }
+
+    foreach ($_FILES as $name => $values) {
+        $fileName = preg_replace('/\s/', '_', $values["name"]);
+        if (file_exists($publicationPath . $fileName)) {
+            showError("File $fileName already exists!");
+        }
+        $extension = pathinfo($fileName)['extension'];
+        $index     = preg_split('/_/', $name)[1];
+
+        if (!isset($extension)) {
+            showError(
+                "Please make sure your file has a valid extension: " .
+                $values['name']
+            );
+        }
+        $pubTypeID       = isset($_REQUEST['publicationType_'.$index]) ?
+            $_REQUEST['publicationType_'.$index] : null;
+        $pubCitation     = isset($_REQUEST['publicationCitation_'.$index]) ?
+            $_REQUEST['publicationCitation_'.$index] : null;
+        $pubVersion      = isset($_REQUEST['publicationVersion_'.$index]) ?
+            $_REQUEST['publicationVersion_'.$index] : null;
+        $pubUploadInsert = array(
+                            'PublicationID'           => $pubID,
+                            'PublicationUploadTypeID' => $pubTypeID,
+                            'URL'                     => $fileName,
```

Comment:
$fileName should be called with `basename()` around it for security purposes. This will give you just the filename and extension and prevent attackers from accessing other files.

If you need a path, you'll need to wrap it with `realpath()`